flytekit/core/type_engine.py

```diff
@@ -147,18 +157,36 @@ def __init__(
         from_literal_transformer: typing.Callable[[Literal], T],
     ):
         super().__init__(name, t)
+        self._type = t
         self._lt = lt
         self._to_literal_transformer = to_literal_transformer
         self._from_literal_transformer = from_literal_transformer
 
     def get_literal_type(self, t: Type[T] = None) -> LiteralType:
-        return self._lt
+        return LiteralType.from_flyte_idl(self._lt.to_flyte_idl())
```

Comment:
it should either be here or the `TypeTransformer` for `Union` must copy all the types before adding the `_structure` field. For most other transformers this is not an issue because they construct a new `LiteralType` instance on each invocation, but these ones reuse the same instance so modifying it poisons all future invocations

Reply:
got it. Let's leave it here.